src/harness/harness.ts

```diff
@@ -23,18 +23,60 @@
 /// <reference path="virtualFileSystem.ts" />
 /// <reference types="node" />
 /// <reference types="mocha" />
-/// <reference types="chai" />
-
 
 // Block scoped definitions work poorly for global variables, temporarily enable var
 /* tslint:disable:no-var-keyword */
 
-// this will work in the browser via browserify
-var _chai: typeof chai = require("chai");
-var assert: typeof _chai.assert = _chai.assert;
-// chai's builtin `assert.isFalse` is featureful but slow - we don't use those features,
-// so we'll just overwrite it as an alterative to migrating a bunch of code off of chai
-assert.isFalse = (expr, msg) => { if (expr as any as boolean !== false) throw new Error(msg); };
+function assert(expr: boolean, msg?: string | (() => string)): void {
+    if (!expr) {
+        throw new Error(typeof msg === "string" ? msg : msg());
+    }
+}
+namespace assert {
+    export function isFalse(expr: boolean, msg = "Expected value to be false."): void {
+        assert(!expr, msg);
+    }
+    export function equal<T>(a: T, b: T, msg = "Expected values to be equal."): void {
+        assert(a === b, msg);
+    }
+    export function notEqual<T>(a: T, b: T, msg = "Expected values to not be equal."): void {
+        assert(a !== b, msg);
+    }
+    export function isDefined(x: {} | null | undefined, msg = "Expected value to be defined."): void {
+        assert(x !== undefined && x !== null, msg);
+    }
+    export function isUndefined(x: {} | null | undefined, msg = "Expected value to be undefined."): void {
+        assert(x === undefined, msg);
+    }
+    export function deepEqual<T>(a: T, b: T, msg?: string): void {
+        assert(isDeepEqual(a, b), msg || (() => `Expected values to be deeply equal:\nExpected:\n${JSON.stringify(a, undefined, 4)}\nActual:\n${JSON.stringify(b, undefined, 4)}`));
+    }
+    export function lengthOf(a: ReadonlyArray<{}>, length: number, msg = "Expected length to match."): void {
+        assert(a.length === length, msg);
+    }
+    export function throws(cb: () => void, msg = "Expected callback to throw"): void {
+        let threw = false;
+        try {
+            cb();
+        }
+        catch {
+            threw = true;
+        }
+        assert(threw, msg);
+    }
+
+    function isDeepEqual<T>(a: T, b: T): boolean {
+        if (a === b) {

```

Comment:
I have no idea if we have any tests which could trigger this (probably not), but `NaN === NaN` is false, but I'd generally say one would expect `{ x: NaN }` to deep equal `{ x: NaN }`; ergo there should probably be a clause `if (a !== a && b !== b) return true;`.  Probably a pointless remark unless we start testing more `NaN`y edgecases.